notebooks/Get_Workflow_Metadata.ipynb

```diff
@@ -0,0 +1,268 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Get Workflow Data\n",
+    "\n",
+    "This notebook uses the Firecloud API to:\n",
+    "1. List all the submissions of the current workspace.\n",
+    "2. List all the workflows of a selected submission.\n",
+    "3. For each workflow, parse out the cost, duration, and other metadata.\n",
+    "\n",
+    "You may run step 1 of this notebook as-is. No modifications are necessary. Step 2 requires you to fill in the BigQuery IDs.\n",
+    "\n",
+    "After running step 1 of this notebook, you will be prompted to select a `submissionId`. **Please copy and paste one from the displayed table, or type in \"all\" (without quotes) to pull all submissions**.\n",
+    "\n",
+    "If your submission has several workflows, the **notebook may take several minutes to run**. This is because we must make one API call per workflow. For example, if your submission has 427 workflows, the notebook will make 427 sequential API calls which may take 4-6 minutes. "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "fapi.__version__"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import firecloud.api as fapi"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import firecloud.api as fapi\n",
+    "import os\n",
+    "import pandas as pd\n",
+    "from datetime import datetime\n",
+    "from ipywidgets import IntProgress\n",
+    "from IPython.display import display\n",
+    "import pandas_gbq\n",
+    "\n",
+    "# Adjust these settings as needed for the dataframe displays\n",
+    "pd.set_option('display.max_rows', 500)\n",
+    "pd.set_option('display.max_colwidth', 300)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Step 1: Load submission data"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "namespace = os.environ['WORKSPACE_NAMESPACE']\n",
+    "workspace = os.environ['WORKSPACE_NAME']\n",
+    "print(f'Current workspace is {namespace}/{workspace}')"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "r = fapi.list_submissions(namespace, workspace)\n",
+    "df = ''\n",
```

Comment:
I'd move this assigment to the `else` block below - it's a bit confusing here

Reply:
Ah - technically your suggestion would work as Python variables are scoped to the who function, not just in the innermost "block" (see https://www.studytonight.com/python-howtos/scope-of-the-variable-initialised-in-the-if-statement). However, given that the df is used outside of the if/else block, it's a good practice to "declare it" outside of the block. This makes it safer from future code changes that modify one branch of the if/else.

Reply:
hm, that's fair. 👍